src/components/shape/ammo-shape.js

```diff
@@ -0,0 +1,82 @@
+/* global Ammo,THREE */
+const threeToAmmo = require("three-to-ammo");
+
+var AmmoShape = {
+  schema: {
+    type: { default: "hull", oneOf: ["box", "cylinder", "sphere", "capsule", "cone", "hull", "mesh"] },
+    recenter: { default: false }, //recenter the object3D's geometry
+    autoGenerateShape: { default: true }, //disable if using custom halfExtents or sphereRadius
```

Comment:
Is this property necessary? It seems like we should be able to infer this, e.g. detect when `halfExtents` or `sphereRadius` are specified.

Reply:
I wanted to do that, but aframe makes it exceedingly difficult. While you can initialize a number to NaN in a component schema, there is no way, afaik, to initialize a vec3 in any state where you can determine with 100% accuracy that it hasn't been set by a user or not.

Any of the following will throw errors:
```
halfExtents: { type: "vec3", default: { x: NaN, y: NaN, z: NaN } }
halfExtents: { type: "vec3", default: { x: null, y: null, z: null} }
halfExtents: { type: "vec3", default: NaN }
halfExtents: { type: "vec3", default: null }
```

I am open to any suggestions that would avoid using an additional schema property, but I haven't yet found one.

Reply:
Ok thanks – suggestion below.